chart/templates/cronjob.yaml

```diff
@@ -26,15 +29,17 @@ spec:
           # If a container fails, fail the pod (job controller will make a fresh new pod)
           restartPolicy: Never
           securityContext:
-            runAsNonRoot: true
+            runAsNonRoot: {{ .Values.runAsNonRoot }}
```

Comment:
Are there any cases where it would need to run as root? It's completely unprivileged.

Reply:
@rptaylor On second inspection this may be an issue pertaining to cluster configuration as well. I'm also receiving an error related to runAsNoonRoot when running the production kapel helm chart in CHTC's kubernetes cluster, since the `library/python:3` image runs as root by default.
```
Error: container has runAsNonRoot and image will run as root (pod: "kapel-test-job", container: processor)
```
The [documentation that I can find](https://kubernetes.io/docs/concepts/security/pod-security-standards/#restricted) pertaining to `securityContext.runAsNonRoot` indicates that the flag just checks whether a pod's containers runs as root and throws an error if they do. Should we add additional configuration (such as [`runAsUser: UID > 0`](https://kubernetes.io/docs/tasks/configure-pod-container/security-context/))?

Reply:
Ah, unfortunately our cluster is a bit old and still using PSP, one of the potential pitfalls of PSP is that it is mutating and can change the default runAsUser. For us it runs as UID 10000 by default. But we should make it explicit in the cronJob spec instead of relying on the default of the image, or any changes that extraneous PSP might apply.

I still think having a fixed value `runAsNonRoot: true` is a good safeguard, and yes let's add variables for runAsUser and runAsGroup. The default UID, GID could be 10000, it's just arbitrary.

Sometimes Openshift needs to configure runAs* so this will help for compatibility too.

Reply:
Just updated the helm chart to return the fixed `runAsNonRoot: true` and make runAsUser/Group configurable. We had a couple issues getting the gratia output image to run with a non-root user since the gratia rpms install with root ownership by default, we were able to selectively chmod things in the Dockerfile to get around that though.

Reply:
Wouldn't the gratia RPM installation happen at the container image build time? But anyway , okay good that it is all working unprivileged.